Update: Ad Portal Template Browser Flow

jennsun opened the pull request:
https://app.clickup.com/t/1bmaz26

Updated TemplateBrowser component to include new features for Ad Portal flow. 
For ad portal flow:
- Updated template browser title to "Choose a template for your commercial"
- Updated subtitle to "All of these premium video templates are fully specced and ready to run on TV."
- Hid '6 seconds' option from Duration filter
- By default, have the templates filtered to only 'wide' only, but hide the Aspect ratio filter menu. 

FOR ALL FLOWS:
- Increased padding between filter name the border from the filter menu above it to 35px
- Added 10px padding below the filter name and filter values

jennsun committed:
Update: Ad portal Template Browser as Two Separate Components

jennsun committed:
Update: Ad Portal Template Browser Flow

simanthastew commented on example_site/src/components/AdPortalLanding.js:
```diff
@@ -14,15 +13,16 @@ function SelectTemplateButton() {
+            const account = await waymarkInstance.getAccountInfo();
+            setAccount(account);
+        } catch (error) {
+            console.error(error);
```
For all of these try/catches, let's put something a little more descriptive with the error message. For example:
console.error('Error creating account: ${error}')

simanthastew replied:
Also, is this function pretty identical to the one we use in the account authentication app?

simanthastew commented on example_site/src/components/App.js:
```diff
@@ -64,7 +64,10 @@ function App() {
-        <TemplateBrowser waymarkInstance={waymarkInstance} />
+        <TemplateBrowser 
+          waymarkInstance={waymarkInstance} 
+          isAdPortalFlow={siteConfiguration.id==='adPortal'}
```
Two quick things:
1. Add spaces between `id`, `===`, and `'adPortal'`
2. Let's create a constant for each of the site configuration IDs that we can reference in the `siteConfiguration` objects themselves and elsewhere in the code

simanthastew commented on example_site/src/components/TemplateBrowser.js:
```diff
@@ -150,9 +150,16 @@ function CollectionTemplates({
 }
 
-export default function TemplateBrowser() {
+export default function TemplateBrowser ({isAdPortalFlow}) {
```
The spacing should look like this - sorry we never got linting setup in this repository!
`TemplateBrowser({ isAdPortalFlow }) {`

simanthastew commented on example_site/src/components/TemplateBrowser.js:
```diff
@@ -150,9 +150,16 @@ function CollectionTemplates({
+      return ({aspectRatio: "16:9", duration: [15, 30]});
+    }
+    return ({});
+  });
```
I think we can make this a little cleaner, and you do not need the surrounding parens in your return values here
```
const [templateFilter, setTemplateFilter] = useState(() =>
    isAdPortalFlow ? {aspectRatio: "16:9", duration: [15, 30]} : {})
```

simanthastew commented on example_site/src/components/TemplateBrowser.js:
```diff
@@ -150,9 +150,16 @@ function CollectionTemplates({
+    }
+    return ({});
+  });
+  const newDurationFilters = isAdPortalFlow ? durationFilters.slice(1) : durationFilters;
```
I think we should target the 6 second filter here specifically in case the order ever changes or we add in additional values! Or probably better yet, target the 15 and 30 second values and then we don't care about what else is in the object.

simanthastew replied:
And let's call this something other than `newDurationFilters` - they're not really new, they're pared down for the site configuration so let's noodle on a name that evokes that!

simanthastew commented on example_site/src/components/TemplateBrowser.js:
```diff
@@ -176,12 +183,21 @@ export default function TemplateBrowser() {
+          them any way that you like."
+        />
+      )}
+      
```
I don't think we need to do this yet because we don't have plans for additional flows with different template browsers but let's talk about what this component would look like if we included some of this data in the site configuration object.

simanthastew reviewed (changes requested):
Looks really good! Just a handful of small changes

jennsun committed:
PR Review
 Constants for siteConfiguration IDs, Filtered duration filters to be 15 and 30 seconds instead of slicing out 6 second option, created templateBrowserHeader component for different flows, cleaned up code
